src/Angor/Client/Pages/View.razor

```diff
@@ -443,107 +451,135 @@
+        TimeLeft = 0,
+        TotalInvestors = 0,
+        TotalRaised = 0,
+        TotalSeeders = 0,
+        FundingProgressPercent = 0
     };
 
-    ProjectStats projectStats = new ProjectStats
-        {
-            Loading = true,
-            TimeLeft = 0,
-            TotalInvestors = 0,
-            TotalRaised = 0,
-            TotalSeeders = 0,
-            FundingProgressPercent = 0,
-        };
-
-    bool founder = false;
-    bool invested = false;
-    bool findInProgress = false;
-    DateTime currentDate = DateTime.UtcNow;
+    bool founder;
+    bool invested;
+    bool findInProgress;
+    readonly DateTime currentDate = DateTime.UtcNow;
 
     private List<string> NostrClients = new();
 
     protected override async Task OnInitializedAsync()
     {
-        NostrClients = _NetworkConfiguration.GetDefaultRelayUrls().Select(_ => _.Url.ToString()).ToList();
-
-        project = storage.GetInvestmentProjects().FirstOrDefault(p => p.ProjectInfo.ProjectIdentifier == ProjectId);
+        projectStats.Loading = true;
+        findInProgress = false;
+        error = string.Empty;
 
-        if (project is InvestorProject findProject)
-        {
-            invested = findProject.InvestedInProject(); // TODO: need to scan for the invested projects when opening on a new browser
-            myProjectExplorerLink = _NetworkConfiguration.GetExplorerUrl().Url + $"/transaction/{findProject.TransactionId}";
-        }
-        else
+        try
         {
-            project = storage.GetFounderProjects().FirstOrDefault(p => p.ProjectInfo.ProjectIdentifier == ProjectId);
+            NostrClients = _NetworkConfiguration.GetDefaultRelayUrls().Select(_ => _.Url.ToString()).ToList();
 
+            project = storage.GetFounderProjects().FirstOrDefault(p => p.ProjectInfo.ProjectIdentifier == ProjectId);
             if (project != null)
             {
                 founder = true;
+                projectStats.Loading = false;
+                SetProjectLinksAndRefreshBalance();
+                return;
             }
-            else
+
+            project = storage.GetInvestmentProjects().FirstOrDefault(p => p.ProjectInfo.ProjectIdentifier == ProjectId);
+            if (project is InvestorProject investorProject)
             {
-                project = SessionStorage.GetProjectById(ProjectId);
+                invested = investorProject.InvestedInProject();
+                myProjectExplorerLink = _NetworkConfiguration.GetExplorerUrl().Url + $"/transaction/{investorProject.TransactionId}";
+                projectStats.Loading = false;
+                SetProjectLinksAndRefreshBalance();
+                return;
+            }
 
-                if (project == null)
-                {
-                    findInProgress = true;
+            project = SessionStorage.GetProjectById(ProjectId);
+            if (project != null)
+            {
+                projectStats.Loading = false;
+                StateHasChanged();
```

Comment:
Shouldn't you be calling the method `SetProjectLinksAndRefreshBalance` here?
I see that the explorer link is missing